### PR TITLE
Fix ability to run "ruff check ." from repo root

### PR DESCRIPTION
**Pull request by ogenstad**

### CODE REVIEW

File: python_sdk/pyproject.toml [hunk 1]
```diff
@@ -308,6 +308,13 @@ max-complexity = 17
...
 
 
+"infrahub_sdk/client.py" = [
+    ##################################################################################################
+    # Review and change the below later                                                              #
+    ##################################################################################################
+    "PLR0904",  # Too many public methods
+]
+
 "python_sdk/infrahub_sdk/client.py" = [
```
Comment (dgarros) [inline python_sdk/pyproject.toml:318]: is the section below also required ? or is it never gonna be used anyway ?

File: python_sdk/pyproject.toml [hunk 2]
```diff
@@ -308,6 +308,13 @@ max-complexity = 17
 [tool.ruff.lint.per-file-ignores]
 
 
+"infrahub_sdk/client.py" = [
+    ##################################################################################################
```
Comment (dgarros) [inline python_sdk/pyproject.toml:312]: Can we get rid of the comment altogether here, it feels like these is nothing more to review :)

**Review by dgarros — APPROVED** [review]

File: python_sdk/pyproject.toml (hunk 1 above)
Comment (ogenstad) [reply python_sdk/pyproject.toml:318]: It is required not that we're also running `ruff check python_sdk --config python_sdk/pyproject.toml` in the CI pipeline with that command the original paths are incorrect.

Until we've actually split the repos I think it's nice to be able to run ruff with just one command and have everything covered.

File: python_sdk/pyproject.toml (hunk 2 above)
Comment (ogenstad) [reply python_sdk/pyproject.toml:312]: It depends :)

I don't know if this is a bug in ruff, it counts the overload as an extra public method which might or might not be correct depending on your philosophy.

We can probably move away some of the public methods, for instance I don't know if .login really needs to be public and perhaps the ip stuff should live elsewhere. But we could also delete the comment for now, if ruff doesn't change this behaviour I don't think we'll get under 20 any time soon.

File: python_sdk/pyproject.toml (hunk 1 above)
Comment (dgarros) [reply python_sdk/pyproject.toml:318]: is this command still needed with this fix ? or can we revert to using `ruff check .` at the root level ?

Comment (ogenstad) [reply python_sdk/pyproject.toml:318]: Spoke to @wvandeun separately about this as the format issue came up elsewhere. I didn't understand why you needed the original change but I think now that it was due to issues with `invoke format` or `invoke sdk.format` is that correct? I think the main problem here might be that the invoke format commands are setup incorrectly in that they target the SDK pyproject.toml files so the paths gets mixed up. Would that be in line with what you saw?

**Comment by ogenstad** [comment]
Closing as this is no longer applicable in its current state with the separation of the repos.

**Pull request closed without merging**